tempesta_fw/http.c

```diff
@@ -2363,6 +2421,11 @@ tfw_http_adjust_req(TfwHttpReq *req)
 	int r;
 	TfwHttpMsg *hm = (TfwHttpMsg *)req;
 
+	/*
+	 * TODO: generate HTTP/1.1 request from collected sequence of
+	 * HEADERS/CONTINUATION/DATA frames' payload (for HTTP/2 mode).
+	 */
```

Comment:
I removed this comment, since it seems now that almost all the HTTP/2 -> HTTP/1.1 transformation work possibly can be implemented in HPACK-decoder/HTTP/2-parser context (in-place, during writing the decoded data into the new skbs). `tfw_http_adjust_req()` procedure may be left unchanged (except of HTTP/2 pseudo-headers replacements).

Reply:
Maybe it'd be most efficient to make HTTP/2 -> HTTP/1.1 transformation on HTTP/2 request decoding, but I'm not sure how well it scales. If we'll have backends supporting HTTP/2 at the same time with pure HTTP/1.1 (it's likely that we'll come to HTTP/2 for backends), won't we need to rewrite the logic because we don't know which backend we have to send a request before we decode it and process with HTTPtables?

Reply:
Yeah, it will be taken into account in the next - HPACK PR.